File: problems/2347-best-poker-hand/solution.py
class Solution:
    def bestHand(self, ranks: List[int], suits: List[str]) -> str:
        if len(set(suits)) == 1:
            return "Flush"
        
        rank_counts = {}
        for rank in ranks:
            rank_counts[rank] = rank_counts.get(rank, 0) + 1
        
        max_count = 0
        for count in rank_counts.values():
            max_count = max(max_count, count)
        
        if max_count >= 3:
            return "Three of a Kind"
        elif max_count == 2:
            return "Pair"
        else:
            return "High Card"

File: problems/2347-best-poker-hand/solution.java
class Solution {
    public String bestHand(int[] ranks, char[] suits) {
        boolean flush = true;
        for (int i = 1; i < 5; i++) {
            if (suits[i] != suits[0]) {
                flush = false;
                break;
            }
        }
        if (flush) {
            return "Flush";
        }

        int[] rankCounts = new int[14];
        for (int rank : ranks) {
            rankCounts[rank]++;
        }

        boolean threeOfAKind = false;
        boolean pair = false;
        for (int count : rankCounts) {
            if (count >= 3) {
                threeOfAKind = true;
                break;
            } else if (count == 2) {
                pair = true;
            }
        }

        if (threeOfAKind) {
            return "Three of a Kind";
        } else if (pair) {
            return "Pair";
        } else {
            return "High Card";
        }
    }
}